Orm/Xtensive.Orm/Orm/QueryResult.cs

```diff
@@ -29,17 +31,20 @@ public EnumerableReader(IEnumerable<TItem> items)
       }
     }
 
-    private readonly IMaterializingReader<TItem> reader;
     private readonly StateLifetimeToken lifetimeToken;
 
+
+    // DO NOT ENUMERATE this reader anywhere outside this class
+    internal IMaterializingReader<TItem> Reader { get; }
```

Comment:
I thought it would be a good idea to keep `reader` as a field but add the internal `Session` property instead.